scripts/adage_pancancer.py

```diff
@@ -0,0 +1,115 @@
+"""
+Gregory Way 2017
+Variational Autoencoder - Pan Cancer
+scripts/adage_pancancer.py
+
+Comparing a VAE learned features to ADAGE features. Use this script within
+the context of a parameter sweep to compare performance across a grid of
+hyper parameters.
+
+Usage:
+
+    Run in command line with required command arguments:
+
+        python scripts/adage_pancancer.py --learning_rate
+                                          --batch_size
+                                          --epochs
+                                          --sparsity
+                                          --noise
+                                          --output_filename
+
+    Typically, arguments to this script are compiled automatically by:
+
+        python scripts/vae_paramsweep.py --parameter_file <parameter-filepath>
+                                         --config_file <configuration-filepath>
+
+Output:
+Loss and validation loss for the specific model trained
+"""
+
+import os
+import argparse
+import numpy as np
+import pandas as pd
+
+from keras.layers import Input, Dense
+from keras.layers.noise import GaussianDropout
+from keras.models import Model
+from keras import optimizers
+from keras.regularizers import l1
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-l', '--learning_rate',
+                    help='learning rate of the optimizer')
+parser.add_argument('-b', '--batch_size',
+                    help='Number of samples to include in each learning batch')
+parser.add_argument('-e', '--epochs',
+                    help='How many times to cycle through the full dataset')
+parser.add_argument('-s', '--sparsity',
+                    help='How much L1 regularization penalty to apply')
+parser.add_argument('-n', '--noise',
+                    help='How much Gaussian noise to add during training')
+parser.add_argument('-f', '--output_filename',
+                    help='The name of the file to store results')
+args = parser.parse_args()
+
+# Set hyper parameters
+learning_rate = float(args.learning_rate)
+batch_size = int(args.batch_size)
+epochs = int(args.epochs)
+sparsity = float(args.sparsity)
+noise = float(args.noise)
+output_filename = args.output_filename
+
+original_dim = 5000
+latent_dim = 100
+
+# Random seed
+seed = int(np.random.randint(low=0, high=10000, size=1))
```

Comment:
Why the completely random seed here?

Reply:
I wanted to ensure that the splitting of training and testing data wasn't the same each time.